ydb/core/tablet_flat/flat_table.cpp

```diff
@@ -1338,6 +1341,11 @@ TCompactionStats TTable::GetCompactionStats() const
     return stats;
 }
 
+void TTable::SetTableObserver(ITableObserverPtr ptr) noexcept
+{
+    TableObserver = std::move(ptr);
```

Comment:
Let's check that current `TableObserver ` is null?

Reply:
Why? I may want to install different observers depending on the lifecycle.

Reply:
It's easy to double call `SetTableObserver` and expect that both observers will work

But if we really need such an option we can additionally have `ResetTableObserver` method, so anyone who call `SetTableObserver` should reset it before a next user

Reply:
I would prefer a simple uncomplicated setter for now, then clients could just call `SetTableObserver(nullptr)` without the need for a separate reset method.

Reply:
`SetTableObserver(nullptr)` is ok instead of  `ResetTableObserver`